webapp/src/components/ImageFieldEditor.tsx

```diff
@@ -197,7 +226,82 @@ export class ImageFieldEditor<U extends pxt.Asset> extends React.Component<Image
         this.galleryAssets = getAssets(true, this.asset.type);
     }
 
-    protected filterAssets(assets: pxt.Asset[], type: pxt.AssetType = this.asset?.type, isGallery = false, useTags?: boolean) {
+    protected getAvailableTags(filterAssets: pxt.Asset[]) {
+        let collectedTags: string[] = [];
+
+        if (this.galleryAssets) {
+            const specialTags = ["tile", "dialog", "background"];
+            filterAssets.forEach( (asset) => {
+                if (asset.meta.tags) {
+                    asset.meta.tags.forEach(t => {
+                        let sanitizedTag = sanitize(t);
+                        if (specialTags.indexOf(sanitizedTag) == -1) {
+                            if (collectedTags.indexOf(sanitizedTag) < 0) {
+                                collectedTags.push(sanitizedTag);
+                            }
+                        }
+                    });
+                }
+            })
+
+            return collectedTags;
+        }
+        return [];
+
+        function sanitize(tag: string) {
+            let sanitizedTag = (tag.indexOf("!") === 0 || tag.indexOf("?") === 0) && tag.length > 1 ? tag.substring(1) : tag;
```

Comment:
probably don't need to trim `!` - `!` is only defined for gallery filters and not image tags so it wouldn't really need to be trimmed here / the tag shouldn't hit here? re: https://makecode.com/cli/buildsprites#tags (I guess no harm leaving though? just an extra that won't be hit in validly defined tags)